lib/json-schema/validator.rb

```diff
@@ -238,6 +242,46 @@ def validation_errors
       @errors
     end
 
+    def resolve_nested_references(schema, schema_portion, list)
```

Comment:
`schema_portion` appears to be a name that's new to the project.  Could you pick a name that's already been used before, so it's easier to work out what it refers to?

Also, it's not clear what `list` is.  Could you find a more suitable name?